pymeasure/instruments/hp/hp3437A.py

```diff
@@ -0,0 +1,372 @@
+        # (cf table 3-2, page 3-5 of the manual, HPAK document 9018-05946)
+        decode_map = {
+           0: math.nan,
+           1: 0.1,
+           2: 10.0,
+           3: 1.0,
+        }
+        return decode_map[r]
+
+    @staticmethod
+    def _decode_trigger(t):
+        """Method to decode trigger mode
+
+        :param status_bytes: list of bytes to be decoded
+        :return trigger_mode: string with the current trigger mode
+        :rtype trigger_mode: str
+
+        """
+        decode_map = {
+           0: "INVALID",
+           1: "internal",
+           2: "external",
+           3: "hold/manual"
+        }
+        return decode_map[t]
+
+    _get_process_ = {
+        "Number": StatusBitsBase._convert_from_bcd,
+        "Delay": StatusBitsBase._convert_from_bcd,
+        "Range": _decode_range,
+        "Trigger": _decode_trigger,
+        }
+
+    def __str__(self):
+        """
+        Returns a pretty formatted string showing the status of the instrument
+        """
+        ret_str = ""
+        for field in self._fields_:
+            ret_str = ret_str + f"{field[0]}: {getattr(self, field[0])}\n"
+
+        return ret_str
+
+
+class PackedBits(ctypes.BigEndianStructure):
+    """
+    A bitfield structure containing the assignments for the data transfer in packed/binary mode
+    """
+    _pack_ = 1
+    _fields_ = [
+        ("range", c_uint8, 2),  # bit 0..1
+        ("sign_bit", c_uint8, 1),
+        ("MSD", c_uint8, 1),
+        ("SSD", c_uint8, 4),
+        ("TSD", c_uint8, 4),
+        ("LSD", c_uint8, 4), ]
+
+    def __float__(self):
+        """
+        Return a float value from the packed data of the HP3437A
+
+        """
+        # range decoding
+        # (cf table 3-2, page 3-5 of the manual, HPAK document 9018-05946)
+        decode_map = {
+           1: 0.1,
+           2: 10.0,
+           3: 1.0,
+        }
+        cur_range = decode_map[self.range]
+
+        signbit = 1
+        if self.sign_bit == 0:
+            signbit = -1
+
+        return (
+            cur_range * signbit * (
+                self.MSD + self.SSD / 10 + self.TSD / 100 + self.LSD / 1000
+            )
+        )
+
+
+class HP3437A(HPLegacyInstrument):
+    """Represents the Hewlett Packard 3737A system voltmeter
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+    status_desc = Status
+    pb_desc = PackedBits
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "Hewlett-Packard HP3437A",
+            status_bitfield=self.status_desc,
+            **kwargs,
+        )
+        log.info("Initialized HP3437A")
+
+    # Definitions for different specifics of this instrument
+    RANGE = {
+        1e-1: "R1",
+        1: "R2",
+        10: "R3",
+    }
+
+    TRIGGERS = {
+        "internal": "T1",
+        "external": "T2",
+        "hold": "T3",
+        "manual": "T3",
+    }
+
+    class SRQ(IntFlag):
```

Comment:
I think this class is not used anywhere. Right ?
Take into account that if you need to further split the field `SRQ`, you could use nested `ctypes` enums.

Reply:
Thanks for the reminder, i was returning only the numeric value of the SRQ mask, now changed to the enum return, as in HP3478A